test/tests/clone.coffee

```diff
@@ -0,0 +1,538 @@
+          'Clone child'
+        ] }
+        'Not a clone'
+      ] }
+    ]
+
+    t.sendKeys 'ggdd'
+    t.expect [
+      { text: 'Not a clone', children: [
+        { text: 'Marked clone', mark: 'mark', children: [
+          'Clone child'
+        ] }
+        'Not a clone'
+      ] }
+    ]
+    t.expectMarks { 'mark': 1 }
+
+    t.sendKeys 'dd'
+    t.expect [ "" ]
+    t.expectMarks { }
+
+  it "works with marks in tricky case 2", () ->
+    t = new TestCase [
+      { text: 'parent', children: [
+        { text: 'Not a clone', children: [
+          { text: 'Marked child', mark: 'mark' }
+        ] }
+        { text: 'blah', children: [
+          'blah'
+        ] }
+      ] }
+    ]
+    t.expectMarks { 'mark': 3 }
+
+    t.sendKeys 'jycGp'
+    t.expect [
+      { text: 'parent', children: [
+        { text: 'Not a clone', children: [
+          { text: 'Marked child', mark: 'mark' }
+        ] }
+        { text: 'blah', children: [
+          'blah'
+          { text: 'Not a clone', children: [
+            { text: 'Marked child', mark: 'mark' }
+          ] }
+        ] }
+      ] }
+    ]
+    t.expectMarks { 'mark': 3 }
+
+    t.sendKeys 'jj'
+    t.sendKeys ['m', 'enter']
+    t.expect [
+      { text: 'parent', children: [
+        { text: 'Not a clone', children: [
+          'Marked child'
+        ] }
+        { text: 'blah', children: [
+          'blah'
+          { text: 'Not a clone', children: [
+            'Marked child'
+          ] }
+        ] }
+      ] }
+    ]
+    t.expectMarks {}
+
+    t.sendKeys 'ggdd'
+    t.expect [
+      ''
+    ]
+    t.expectMarks {}
+
+    t.sendKeys 'u'
+    t.expect [
+      { text: 'parent', children: [
+        { text: 'Not a clone', children: [
+          'Marked child'
+        ] }
+        { text: 'blah', children: [
+          'blah'
+          { text: 'Not a clone', children: [
+            'Marked child'
+          ] }
+        ] }
+      ] }
+    ]
+    t.expectMarks {}
+
+  it "works with marks in tricky case 3", () ->
+    t = new TestCase [
+      { text: 'parent', children: [
+        { text: 'Not a clone', children: [
+          { text: 'Marked child', mark: 'mark' }
+        ] }
+        { text: 'blah', children: [
+          'blah'
+        ] }
+      ] }
+    ]
+    t.expectMarks { 'mark': 3 }
+
+    t.sendKeys 'jycGp'
+    t.expect [
+      { text: 'parent', children: [
+        { text: 'Not a clone', children: [
+          { text: 'Marked child', mark: 'mark' }
+        ] }
+        { text: 'blah', children: [
+          'blah'
+          { text: 'Not a clone', children: [
+            { text: 'Marked child', mark: 'mark' }
+          ] }
+        ] }
+      ] }
+    ]
+    t.expectMarks { 'mark': 3 }
+
+    t.sendKeys 'jjdd'
+    t.expect [
+      { text: 'parent', children: [
+        'Not a clone'
+        { text: 'blah', children: [
+          'blah'
+          'Not a clone'
+        ] }
+      ] }
+    ]
+    t.expectMarks {}
```

Comment:
are you sure it's wrong?  I understand marks are a per-id feature.  but all instances of that id disappeared, right?  that final t.expect passes but the expectMarks doesn't, which at the very least seems inconsistent

sorry for the bad names for the rows